jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryClientTest.java

```diff
@@ -211,6 +212,37 @@ public void testConfigureBasicAuth()
         registry.getInputRead(), CoreMatchers.containsString("Authorization: Basic dXNlcjpwYXNz"));
   }
 
+  @Test
+  public void testPullManifest()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException,
+          RegistryException {
+    String manifestResponse =
+        "HTTP/1.1 200 OK\nContent-Length: 307\n\n{\n"
+            + "    \"schemaVersion\": 2,\n"
+            + "    \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "    \"config\": {\n"
+            + "        \"mediaType\": \"application/vnd.docker.container.image.v1+json\",\n"
+            + "        \"size\": 7023,\n"
+            + "        \"digest\": \"sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7\"\n"
+            + "    }\n"
+            + "}";
+
+    registry = new TestWebServer(false, Arrays.asList(manifestResponse), 1);
+    RegistryClient registryClient = createRegistryClient(null);
+    ManifestAndDigest<?> manifestAndDigest = registryClient.pullManifest("image-tag");
+
+    // verify details of the returned manifest
```

Comment:
```suggestion
    Assert.assertTrue(manifestAndDigest.getManifest() instanceof V22ManifestTemplate);
```
Check the instance here before force-casting below so that in case it's not of `V22ManifestTemplate`, the test doesn't crash with `ClassCastException`.

And we can skip trivial comments.

Reply:
Done !! This a clever way of avoiding type an object which is not of `V22ManifestTemplate`.